lib/block_view_stake.go

```diff
@@ -2285,6 +2339,81 @@ func (bav *UtxoView) GetStakeEntriesForValidatorPKID(validatorPKID *PKID) ([]*St
 	return stakeEntries, nil
 }
 
+func (bav *UtxoView) GetTopStakesByStakeAmount(limit uint64) ([]*StakeEntry, error) {
+	// Validate limit param.
+	if limit == uint64(0) {
+		return []*StakeEntry{}, nil
+	}
+
+	// Create a slice of UtxoViewStakeEntries. We want to skip pulling these from the database in
+	// case they have been updated in the UtxoView and the changes have not yet flushed to the database.
+	// Updates to a StakeEntry could include adding/removing stake or being deleted which would
+	// impact our ordering. We pull N StakeEntry not present in the UtxoView from the database
+	// then sort the UtxoViewStakeEntry and DatabaseStakeEntry together to find the top N
+	// StakeEntry by stake across both the UtxoView and database.
+	var utxoViewStakeEntries []*StakeEntry
+	for _, stakeEntry := range bav.StakeMapKeyToStakeEntry {
+		utxoViewStakeEntries = append(utxoViewStakeEntries, stakeEntry)
+	}
+
+	// Pull top N StakeEntries from the database (not present in the UtxoView).
+	// Note that we will skip stakers that are present in the view because we pass
+	// utxoViewStakeEntries to the function.
+	dbStakeEntries, err := DBGetTopStakesByStakeAmount(bav.Handle, bav.Snapshot, limit, utxoViewStakeEntries)
+	if err != nil {
+		return nil, errors.Wrapf(err, "UtxoView.GetTopStakesByStakeAmount: error retrieving stake entries from db: ")
+	}
+
+	// Cache top N StakeEntries from the db in the UtxoView.
+	for _, stakeEntry := range dbStakeEntries {
+		// We only pull StakeEntries from the db that are not present in the
+		// UtxoView. As a sanity check, we double-check that the StakeEntry
+		// is not already in the UtxoView here.
+		stakeMapKey := StakeMapKey{ValidatorPKID: *stakeEntry.ValidatorPKID, StakerPKID: *stakeEntry.StakerPKID}
+		if _, exists := bav.StakeMapKeyToStakeEntry[stakeMapKey]; !exists {
+			bav._setStakeEntryMappings(stakeEntry)
+		}
+	}
+
+	// Pull !isDeleted, StakeEntries from the UtxoView.
+	var stakeEntries []*StakeEntry
+	for _, stakeEntry := range bav.StakeMapKeyToStakeEntry {
+		if !stakeEntry.isDeleted && !stakeEntry.StakeAmountNanos.IsZero() {
+			stakeEntries = append(stakeEntries, stakeEntry)
+		}
+	}
+
+	// Sort the StakeEntries by StakeAmountNanos DESC.
+	sort.SliceStable(stakeEntries, func(ii, jj int) bool {
```

Comment:
okay so I don't know if we need SliceStable here since we aren't guaranteed to stably create `stakeEntries` since our map iteration won't be consistent. Basically, we should never have two items that are equal based on your less function. Not sure if there's a performance benefit to slice (pattern-defeating quick sort) vs slice stable

Reply:
You're right. There's no need to use SliceStable. We can just use the default sorting function instead.